app/web/src/store/secrets.store.ts

```diff
@@ -130,6 +135,49 @@ export function useSecretsStore() {
             },
           );
         },
+        secretsByLastCreated(state): SecretsOrderedArray {
+          const out = [] as SecretsOrderedArray;
+
+          for (const [key, value] of Object.entries(
+            state.secretDefinitionByDefinitionId,
+          )) {
+            let change: string | undefined;
+            value.secrets.forEach((secret) => {
+              if (!change) change = secret.createdInfo.timestamp;
+              if (new Date(secret.createdInfo.timestamp) > new Date(change)) {
+                change = secret.createdInfo.timestamp;
+              }
+              if (
+                secret.updatedInfo &&
+                new Date(secret.updatedInfo.timestamp) > new Date(change)
+              ) {
+                change = secret.updatedInfo.timestamp;
+              }
+            });
+
+            out.push({
+              id: key,
+              latestChange: change,
+              secrets: value.secrets,
+            });
+          }
+
+          out.sort((a, b) => {
+            if (!a.latestChange && !b.latestChange) return 0;
```

Comment:
Will we ever have secrets with undefined changes?

Reply:
Yes - a secret definition that has been created but no individual secrets have been added yet will have undefined for the latestChange value.